dashboard/app/models/user_level.rb

```diff
@@ -26,6 +26,7 @@ class UserLevel < ActiveRecord::Base
   belongs_to :user
   belongs_to :level
   belongs_to :script
+  belongs_to :level_source
```

Comment:
Yes, this is very dangerous because it gives the impression that `ul.level_source` will work.  It usually will, but for old records it will be nil.

(However `ul.user.last_attempt(ul.level).try(:level_source)` will return the actual attempt.)

Instead of `belongs_to :level_source` maybe we should add a helper method to `UserLevel` with plans to switch to the column once it's backfilled?